### ISSUE TITLE
Migration issues with Laravel 5.2

### ISSUE THREAD

**Issue by anthonyholmes** [Blogify/Blogify#20]
I am getting this error when running `php artisan blogify:migrate` with a fresh install of Laravel 5.2

```
[ErrorException]
Array to string conversion
```

**Comment by jorenvh** [comment]
Do you have a stack trace of the error?

**Comment by ericvanjohnson** [comment]
@jorenvh ran into the same issue here is the trace 

```
[2016-06-05 19:03:24] local.ERROR: ErrorException: Array to string conversion in /home/vagrant/gitrepos/DiegoDevRepos/diegodev-site/diegodev-digitalocean/vendor/jorenvanhocht/blogify/database/Migrations/2015_03_03_212000_create_users_table.php:105
Stack trace:
#0 /home/vagrant/gitrepos/DiegoDevRepos/diegodev-site/diegodev-digitalocean/vendor/jorenvanhocht/blogify/database/Migrations/2015_03_03_212000_create_users_table.php(105): Illuminate\Foundation\Bootstrap\HandleExceptions->handleError(8, 'Array to string...', '/home/vagrant/g...', 105, Array)
#1 /home/vagrant/gitrepos/DiegoDevRepos/diegodev-site/diegodev-digitalocean/vendor/laravel/framework/src/Illuminate/Database/Schema/Builder.php(146): CreateUsersTable->{closure}(Object(Illuminate\Database\Schema\Blueprint))
#2 /home/vagrant/gitrepos/DiegoDevRepos/diegodev-site/diegodev-digitalocean/vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php(219): Illuminate\Database\Schema\Builder->create('users', Object(Closure))
#3 /home/vagrant/gitrepos/DiegoDevRepos/diegodev-site/diegodev-digitalocean/vendor/jorenvanhocht/blogify/database/Migrations/2015_03_03_212000_create_users_table.php(115): Illuminate\Support\Facades\Facade::__callStatic('create', Array)
#4 /home/vagrant/gitrepos/DiegoDevRepos/diegodev-site/diegodev-digitalocean/vendor/jorenvanhocht/blogify/database/Migrations/2015_03_03_212000_create_users_table.php(26): CreateUsersTable->createUsersTable()
#5 /home/vagrant/gitrepos/DiegoDevRepos/diegodev-site/diegodev-digitalocean/vendor/laravel/framework/src/Illuminate/Database/Migrations/Migrator.php(147): CreateUsersTable->up()
#6 /home/vagrant/gitrepos/DiegoDevRepos/diegodev-site/diegodev-digitalocean/vendor/laravel/framework/src/Illuminate/Database/Migrations/Migrator.php(117): Illuminate\Database\Migrations\Migrator->runUp('2015_03_03_2120...', 2, false)
#7 /home/vagrant/gitrepos/DiegoDevRepos/diegodev-site/diegodev-digitalocean/vendor/laravel/framework/src/Illuminate/Database/Migrations/Migrator.php(86): Illuminate\Database\Migrations\Migrator->runMigrationList(Array, Array)
#8 /home/vagrant/gitrepos/DiegoDevRepos/diegodev-site/diegodev-digitalocean/vendor/laravel/framework/src/Illuminate/Database/Console/Migrations/MigrateCommand.php(76): Illuminate\Database\Migrations\Migrator->run('/home/vagrant/g...', Array)
#9 [internal function]: Illuminate\Database\Console\Migrations\MigrateCommand->fire()
#10 /home/vagrant/gitrepos/DiegoDevRepos/diegodev-site/diegodev-digitalocean/vendor/laravel/framework/src/Illuminate/Container/Container.php(507): call_user_func_array(Array, Array)
#11 /home/vagrant/gitrepos/DiegoDevRepos/diegodev-site/diegodev-digitalocean/vendor/laravel/framework/src/Illuminate/Console/Command.php(169): Illuminate\Container\Container->call(Array)
#12 /home/vagrant/gitrepos/DiegoDevRepos/diegodev-site/diegodev-digitalocean/vendor/symfony/console/Command/Command.php(256): Illuminate\Console\Command->execute(Object(Symfony\Component\Console\Input\ArrayInput), Object(Illuminate\Console\OutputStyle))
#13 /home/vagrant/gitrepos/DiegoDevRepos/diegodev-site/diegodev-digitalocean/vendor/laravel/framework/src/Illuminate/Console/Command.php(155): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArrayInput), Object(Illuminate\Console\OutputStyle))
#14 /home/vagrant/gitrepos/DiegoDevRepos/diegodev-site/diegodev-digitalocean/vendor/laravel/framework/src/Illuminate/Console/Command.php(185): Illuminate\Console\Command->run(Object(Symfony\Component\Console\Input\ArrayInput), Object(Illuminate\Console\OutputStyle))
#15 /home/vagrant/gitrepos/DiegoDevRepos/diegodev-site/diegodev-digitalocean/vendor/jorenvanhocht/blogify/src/Commands/BlogifyMigrateCommand.php(50): Illuminate\Console\Command->call('migrate', Array)
#16 [internal function]: jorenvanhocht\Blogify\Commands\BlogifyMigrateCommand->fire()
#17 /home/vagrant/gitrepos/DiegoDevRepos/diegodev-site/diegodev-digitalocean/vendor/laravel/framework/src/Illuminate/Container/Container.php(507): call_user_func_array(Array, Array)
#18 /home/vagrant/gitrepos/DiegoDevRepos/diegodev-site/diegodev-digitalocean/vendor/laravel/framework/src/Illuminate/Console/Command.php(169): Illuminate\Container\Container->call(Array)
#19 /home/vagrant/gitrepos/DiegoDevRepos/diegodev-site/diegodev-digitalocean/vendor/symfony/console/Command/Command.php(256): Illuminate\Console\Command->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#20 /home/vagrant/gitrepos/DiegoDevRepos/diegodev-site/diegodev-digitalocean/vendor/laravel/framework/src/Illuminate/Console/Command.php(155): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#21 /home/vagrant/gitrepos/DiegoDevRepos/diegodev-site/diegodev-digitalocean/vendor/symfony/console/Application.php(791): Illuminate\Console\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#22 /home/vagrant/gitrepos/DiegoDevRepos/diegodev-site/diegodev-digitalocean/vendor/symfony/console/Application.php(186): Symfony\Component\Console\Application->doRunCommand(Object(jorenvanhocht\Blogify\Commands\BlogifyMigrateCommand), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#23 /home/vagrant/gitrepos/DiegoDevRepos/diegodev-site/diegodev-digitalocean/vendor/symfony/console/Application.php(117): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#24 /home/vagrant/gitrepos/DiegoDevRepos/diegodev-site/diegodev-digitalocean/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(107): Symfony\Component\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#25 /home/vagrant/gitrepos/DiegoDevRepos/diegodev-site/diegodev-digitalocean/artisan(35): Illuminate\Foundation\Console\Kernel->handle(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#26 {main}
```

**Comment by jorenvh** [comment]
@ericvanjohnson thanks for the stacktrace I take a look at it!

**Comment by jorenvh** [comment]
Hi @ericvanjohnson & @anthonyholmes,

I tried debugging the issue but i can't reproduce it. Was one of you able to debug it at your dev machines and if so could one of you create a pull request?

I attached a screen of my terminal.  

**What did I do?**
- clean install
- migrated the laravel migration with php artisan:migrate
- migrated the blogify migrations with php artisan blogify:migrate

I did this over and over again but can't reproduce it. I also ran the blogify migrate command without running the laravel migrate command (so on a empty db) but that also worked like expected.

**Comment by ericvanjohnson** [comment]
@jorenvh I honestly wasn't too concerned about it. I was going to try the package out and when it failed I was going to just move on but noticed this issue was opened and figured I would contribute the stack trace I had. However, I would like to see this working so if I have some time this week I will see if I can debug it. Probably something simple, just need to take some time and look into it. Will let you know if I find anything.

**Comment by jorenvh** [comment]
@ericvanjohnson big thanks for sharing & caring, I hope you find the time to take a look at it. I did it my self but because I can't reproduce the error it is hard solving it. Thanks a lot!

**Comment by ericvanjohnson** [comment]
@jorenvh just a little more follow up on this. I record a podcast called [PHPUgly](https://soundcloud.com/phpugly/) and we discussed this Issue on the show (Episode 14 which won't be published until Sunday) and one of the developers on the show thought the issue was more likely related to the PHP version and not an issue with Laravel. He thinks it might be related to a change in PHP 5.4 `Function array dereferencing has been added, e.g. foo()[0].`

He also seems to think the patch I submitted would probably be the best approach to keep backward compatibility. Just something to think about.
